ethsigner/commandline/src/main/java/tech/pegasys/ethsigner/config/tls/client/PicoCliClientTlsOptions.java

```diff
@@ -64,4 +69,49 @@ public boolean isCaAuthEnabled() {
   public Optional<Path> getKnownServersFile() {
     return Optional.ofNullable(knownServersFile);
   }
+
+  public boolean isTlsEnabled() {
+    return tlsEnabled;
+  }
+
+  public String validationMessage() {
+    if (!tlsEnabled) {
+      return dependentOptionSpecifiedMessage();
+    }
+
+    return keyStoreConfigurationValidationMessage() + knownServerFileAndCaValidationMessage();
+  }
+
+  private String dependentOptionSpecifiedMessage() {
+    if (knownServersFile != null
+        || keyStoreOptions.getKeyStoreFile() != null
+        || keyStoreOptions.getPasswordFile() != null) {
+      return "Missing required argument(s): '--downstream-http-tls-enabled=true'\n";
+    }
+    return "";
+  }
+
+  private String knownServerFileAndCaValidationMessage() {
+    if (knownServersFile == null && !caAuthEnabled) {
+      return "Missing required argument(s): '--downstream-http-tls-known-servers-file' must be specified if '--downstream-http-tls-ca-auth-enabled=false'\n";
+    }
+    return "";
+  }
+
+  private String keyStoreConfigurationValidationMessage() {
+    final List<String> missingOptions = new ArrayList<>();
+
+    // ArgGroup custom validation
+    if (keyStoreOptions.getKeyStoreFile() == null && keyStoreOptions.getPasswordFile() != null) {
```

Comment:
maybe make this easier to read (also maybe not) a few ideas:
1. have 2 functions "private boolean isMissing(Object o) { return o == null }" & private boolean isSpecified(Object o) ..."
2. Get the keystoreFile and password file as local variables
3. if(isSpecified(keyStoreFile) != isSpecified(passwordFile)) { error = "You must specify both keystore and password together"}

🤷‍♂️  not sure